src/ig/IGExporter.ts

```diff
@@ -882,9 +882,13 @@ export class IGExporter {
       pkgResource instanceof InstanceDefinition &&
       pkgResource._instanceMeta.usage === 'Example'
     ) {
-      const exampleUrl = pkgResource.meta?.profile?.find(url => this.pkg.fish(url, Type.Profile));
+      const exampleUrl = pkgResource.meta?.profile?.find(url => {
+        const [baseUrl, version] = url.split('|', 2);
+        const availableProfile = this.pkg.fish(baseUrl, Type.Profile);
+        return version == null || version === availableProfile.version;
```

Comment:
I think this should also be checking that `availableProfile` is defined -- otherwise this will return `true` in some cases we don't want it to.

E.g.
```
Profile: MyPatient
Parent: Patient
Description: "An example profile of the Patient resource."
* ^version = "0.1.0"
* name 1..* MS

Instance: PatientExampleWrongMeta
InstanceOf: MyPatient
Description: "An example of a patient with a license to krill."
* meta.profile[+] = "http://example.org/StructureDefinition/MyPatient|1.2.3"
* meta.profile[+] = "http://example.org/StructureDefinition/NotMyPatient"
* name
  * given[0] = "James"
  * family = "Pond"
```

This results in an IG resource w/ incorrect `exampleCanonical`:
```json
{
  "reference": {
    "reference": "Patient/PatientExampleWrongMeta"
  },
  "name": "PatientExampleWrongMeta",
  "description": "An example of a patient with a license to krill.",
  "exampleCanonical": "http://example.org/StructureDefinition/NotMyPatient"
}
```